### PR TITLE
[SPARK-12450][MLLib] Un-persist broadcasted variables in KMeans

### PR DESCRIPTION
**Pull request by rnowling**
SPARK-12450 . Un-persist broadcasted variables in KMeans.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala [hunk 1]
```diff
@@ -353,6 +356,7 @@ class KMeans private (
         ((r, KMeans.findClosest(bcCenters.value(r), p)._1), 1.0)
       }
     }.reduceByKey(_ + _).collectAsMap()
+    bcCenters.unpersist(blocking = false)
```
Comment (thunterdb) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala]: small nit: add new lines around this line

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala [hunk 2]
```diff
@@ -324,6 +326,7 @@ class KMeans private (
             s0
           }
         )
+      bcNewCenters.unpersist(blocking = false)
```
Comment (thunterdb) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala]: small nit: add new lines around this line

**Comment by thunterdb** [comment]
@rnowling thanks for fixing this, I just have some style comments.

**Comment by rnowling** [comment]
@thunterdb Thanks for the review!  Let me know if these changes are what you had in mind.

**Comment by srowen** [comment]
LGTM

**Comment by jkbradley** [comment]
LGTM too
I'll merge this with master, branch-1.6 and branch-1.5

**Comment by rnowling** [comment]
Thanks everyone!

**Pull request closed without merging**